### PR TITLE
Fix #1536: Support ECDSA signature

### LINKED ISSUE #1536: Support ECDSA signature
Currently, we only have an endpoint for ECDSA verification (client signs, server verifies) but not vice versa.

### PR DESCRIPTION
**Pull request by petrdvorak**

### CODE REVIEW

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/controller/api/EcdsaSignatureController.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * @author Petr Dvorak, petr@wultra.com
+ */
+@RestController("ecdsaSignatureController")
```
Comment (banterCZ) [inline powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/controller/api/EcdsaSignatureController.java:39]: Why do we have an explicit name?

File: powerauth-client-model/src/main/java/com/wultra/security/powerauth/client/model/request/SignECDSARequest.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+package com.wultra.security.powerauth.client.model.request;
+
+import lombok.Data;
+
+/**
+ * @author Petr Dvorak, petr@wultra.com
+ */
+@Data
+public class SignECDSARequest {
+    private String activationId;
```
Comment (banterCZ) [inline powerauth-client-model/src/main/java/com/wultra/security/powerauth/client/model/request/SignECDSARequest.java:28]: Don't we want to add the bean validation?

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/controller/api/EcdsaSignatureController.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+     * Calculate ECDSA signature.
+     *
+     * @param request Calculate ECDSA signature request.
+     * @return Calculated ECDSA signature response.
+     * @throws Exception In case the service throws exception.
+     */
+    @PostMapping("/sign")
+    public ObjectResponse<SignECDSAResponse> signDataWithECDSA(@RequestBody ObjectRequest<SignECDSARequest> request) throws Exception {
+        logger.info("SignECDSARequest received: {}", request);
+        final ObjectResponse<SignECDSAResponse> response = new ObjectResponse<>("OK", asymmetricSignatureService.signDataWithECDSA(request.getRequestObject()));
```
Comment (banterCZ) [inline powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/controller/api/EcdsaSignatureController.java:62]: `OK` status code is implicit.
```suggestion
        final ObjectResponse<SignECDSAResponse> response = new ObjectResponse<>(asymmetricSignatureService.signDataWithECDSA(request.getRequestObject()));
```

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/controller/api/EcdsaSignatureController.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+     * Validate ECDSA signature.
+     *
+     * @param request Verify ECDSA signature request.
+     * @return Verify ECDSA signature response.
+     * @throws Exception In case the service throws exception.
+     */
+    @PostMapping("/verify")
+    public ObjectResponse<VerifyECDSASignatureResponse> verifyECDSASignature(@RequestBody ObjectRequest<VerifyECDSASignatureRequest> request) throws Exception {
+        logger.info("VerifyECDSASignatureRequest received: {}", request);
+        final ObjectResponse<VerifyECDSASignatureResponse> response = new ObjectResponse<>("OK", asymmetricSignatureService.verifyECDSASignature(request.getRequestObject()));
```
Comment (banterCZ) [inline powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/controller/api/EcdsaSignatureController.java:77]:
```suggestion
        final ObjectResponse<VerifyECDSASignatureResponse> response = new ObjectResponse<>(asymmetricSignatureService.verifyECDSASignature(request.getRequestObject()));
```

**Review by banterCZ — APPROVED** [review]
Looks good.

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/controller/api/EcdsaSignatureController.java (hunk 3 above)
Comment (petrdvorak) [reply powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/controller/api/EcdsaSignatureController.java:62]: Let's fix this alltogether.

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/controller/api/EcdsaSignatureController.java (hunk 4 above)
Comment (petrdvorak) [reply powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/controller/api/EcdsaSignatureController.java:77]: Let's fix this alltogether.

**Review by banterCZ — APPROVED** [review]

**Pull request merged**